client/components/AddTestToQueueWithConfirmation/index.jsx

```diff
@@ -66,7 +66,13 @@ function AddTestToQueueWithConfirmation({
   );
 
   const existingTestPlanReports =
-    existingTestPlanReportsData?.existingTestPlanVersion?.testPlanReports;
+    existingTestPlanReportsData?.existingTestPlanVersion?.testPlanReports.filter(
+      tpr =>
+        tpr.at?.id === at?.id &&
+        tpr.browser?.id === browser?.id &&
+        tpr.minimumAtVersion?.id === minimumAtVersion?.id &&
+        tpr.exactAtVersion?.id === exactAtVersion?.id
```

Comment:
To the best of my understanding, only one of these may exist at a time so conditionally checking may be good, with `exactAtVersion` being the priority. Does that change your thoughts on this comparison?

I suppose it shouldn't matter though, since in the instance of the one (exact or minimum) that doesn't exist, `undefined === undefined` will be evaluated to true anyways anyways.

Reply:
Yeah I think it still makes sense because we are ensuring the Reports are a match and so if one has an `undefined` minimum then the other would as well. One or the other is going to be a useless check since having only a minimum or a required is probably enforced elsewhere but I think it is a lower cognitive load to read than doing an `OR` between those two.

Reply:
Fair enough on the cognitive load, this is fine with me.

Reply:
Thank you!